pkg/reconciler/v1alpha1/taskrun/resources/input_resource_test.go

```diff
@@ -0,0 +1,313 @@
+/*
+Copyright 2018 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	v1alpha1 "github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	fakeclientset "github.com/knative/build-pipeline/pkg/client/clientset/versioned/fake"
+	informers "github.com/knative/build-pipeline/pkg/client/informers/externalversions"
+	listers "github.com/knative/build-pipeline/pkg/client/listers/pipeline/v1alpha1"
+	"github.com/knative/build-pipeline/pkg/logging"
+	buildv1alpha1 "github.com/knative/build/pkg/apis/build/v1alpha1"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	pipelineResourceLister listers.PipelineResourceLister
+	logger                 *zap.SugaredLogger
+)
+
+func setUp() {
+	logger, _ = logging.NewLogger("", "")
+	fakeClient := fakeclientset.NewSimpleClientset()
+	sharedInfomer := informers.NewSharedInformerFactory(fakeClient, 0)
+	pipelineResourceInformer := sharedInfomer.Pipeline().V1alpha1().PipelineResources()
+	pipelineResourceLister = pipelineResourceInformer.Lister()
+
+	res := &v1alpha1.PipelineResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "workspace",
+			Namespace: "marshmallow",
+		},
+		Spec: v1alpha1.PipelineResourceSpec{
+			Type: "git",
+			Params: []v1alpha1.Param{
+				v1alpha1.Param{
+					Name:  "Url",
+					Value: "https://github.com/grafeas/kritis",
+				},
+			},
+		},
+	}
+	resWithServiceAccount := &v1alpha1.PipelineResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "workspace-sa",
+			Namespace: "marshmallow",
+		},
+		Spec: v1alpha1.PipelineResourceSpec{
+			Type: "git",
+			Params: []v1alpha1.Param{
+				v1alpha1.Param{
+					Name:  "Url",
+					Value: "https://github.com/grafeas/kritis",
+				},
+				v1alpha1.Param{
+					Name:  "ServiceAccount",
+					Value: "kritis-service-account",
+				},
+			},
+		},
+	}
+	pipelineResourceInformer.Informer().GetIndexer().Add(res)
```

Comment:
nice! i will try to add the mock objects [here](https://github.com/knative/build-pipeline/blob/master/pkg/reconciler/v1alpha1/pipelinerun/pipelinerun_test.go#L108) instead of using mocks